specs/mgt-search-suggestion.md

```diff
@@ -0,0 +1,161 @@
+
+| Data type | Data context | Description |
+| --- | --- | --- |
+| default | null: no data | The template used to override the rendering of the entire component.
+| loading | null: no data | The template used to render the state of picker while request to graph is being made. |
+| error | null: no data | The template used if user search returns no users. |
+| no-data | null: no data | An alternative template used if user search returns no users. |
+| search-suggestion-input | null: no data | The template to render search input box. |
+| search-suggestion-label-people | null: no data | The template to render people entity label. |
+| search-suggestion-label-text | null: no data | The template to render text entity label. |
+| search-suggestion-label-file | null: no data | The template to render file entity label. |
+| search-suggestion-people | suggestionPeople[]: The people suggestion details list | The template to render people entity. |
+| search-suggestion-text | suggestionText[]: The text suggestion details list | The template to render text entity. |
+| search-suggestion-file | suggestionFile[]: The file suggestion details list | The template to render file entity. |
+
+The following examples shows how to use the `error` template.
+
+```html
+<mgt-search-suggestion>
+    <template data-type="search-suggestion-texts">
+        <div  data-for="text in texts">
+        {{text.text}}
+        </div>
+    </template>
+</mgt-search-suggestion>
+
+```
+
+## Proposed Solution
+
+### Example 1: basic usage without any callback function
+```<mgt-search-suggestion></mgt-search-suggestion>```
+
+### Example 2: Developer provides a site-id and item-id
+```<mgt-search-suggestion id="search-suggestion"> </mgt-search-suggestion>```
+```
+#User can get this component then bind callback action to this component 
+
+function onClickCallback(suggestionValue) {
+    console.log('suggestion value:', suggestionValue);
+    var searchValue = getSuggestionValue(suggestionValue);
+    window.location.assign('https://www.bing.com/search?q=' + searchValue);
+}
+
+function onEnterKeyPressCallback(originalValue, selectedSuggestionValue) {
+    console.log('original value:', originalValue);
+    console.log('suggestion value:', selectedSuggestionValue);
+    var searchValue = getSuggestionValue(selectedSuggestionValue);
+    window.location.assign('https://www.bing.com/search?q=' + searchValue);
+}
+
+function getSuggestionValue(suggestionValue) {
+    var searchValue = '';
+    if (suggestionValue.entity == 'File') {
+        searchValue = suggestionValue.name;
+    } else if (suggestionValue.entity == 'Text') {
+        searchValue = suggestionValue.text;
+    } else if (suggestionValue.entity == 'People') {
+        searchValue = suggestionValue.displayName;
+    }
+    return searchValue;
+}
+
+var obj = document.getElementById('search-suggestion');
+obj.onClickCallback = onClickCallback;
+obj.onEnterKeyPressCallback = onEnterKeyPressCallback;
```

Comment:
I suggest using events instead of callbacks. Events are more flexible and follow the component model for events.

Ex:

```js
const handleSuggestionClicked = (e) => {
  const suggestion = e.detail;
  const searchValue = getSuggestionValue(suggestionValue);
  window.location.assign('https://www.bing.com/search?q=' + searchValue);
}

var searchBox= document.getElementById('search-suggestion');
searchBox.addEventListener('suggestionClick', handleSuggestionClicked);
searchBox.removeEventListener('suggestionClick', handleSuggestionClicked);
```

Please add a table for `Events`

Reply:
I will try, thanks

Reply:
Resolved, added two event, onEnterPress and onEntityClick